R/derive_param_tte.R

```diff
@@ -450,6 +447,13 @@ derive_param_tte <- function(dataset = NULL,
     new_param <- select(new_param, !!!negate_vars(by_vars))
   }
 
+  # check newly created parameter(s) do not already exist
+  if (!is.null(set_values_to$PARAMCD) && !is.null(dataset)) {
+    for (i in seq_along(new_param$PARAMCD)) {
```

Comment:
I would restrict `new_params$PARAMCD` to unique values (`unique_params <- unique(new_params$PARAMCD)`). Otherwise each `PARAMCD` value is checked several times.

Reply:
Done.